src/app/course/course-question-snippet/course-question-snippet.component.ts

```diff
@@ -80,4 +93,73 @@ export class CourseQuestionSnippetComponent implements OnInit {
         }
         return '';
     }
+
+    deleteReport(): void {
+        this.questionReportService.deleteReport(this.reportUQJ.report.id).subscribe(() => {
+            this.notificationsService
+                .show('The Report has been Deleted Successfully.', {
+                    status: TuiNotification.Success
+                }).subscribe();
+            this.reportUQJ.report = {} as QuestionReport;
+        });
+    }
+
+    fillReport(): void {
+        this.formGroup.setValue({
+            description: this.reportUQJ.report.report,
+            description_text: this.reportUQJ.report.report_details
+        });
+    }
+
+    createModal(content: unknown, uqj: UQJ): void {
+        this.reportUQJ = uqj;
+        this.formGroup.reset();
+        if (this.reportUQJ.report.id != null) {
+            this.modalService.open(content, {ariaLabelledBy: 'modal-basic-title', centered: true});
+            this.fillReport();
+        } else {
+            this.modalService.open(content, {ariaLabelledBy: 'modal-basic-title', centered: true});
+        }
+    }
+
+    reportQuestion(): void {
+        if(this.reportUQJ.report.id != null){
+            const data: {report: string, report_details: string } = {
+                report: this.formGroup.get('description').value,
+                report_details: this.formGroup.get('description_text').value
+            };
+            this.questionReportService.updateReport(data, this.reportUQJ.report.id).subscribe(response => {
+                this.notificationsService
+                    .show('The Report was been Created Successfully.', {
+                        status: TuiNotification.Success
+                    }).subscribe();
+                this.reportUQJ.report = response;
+            });
+        }
+        else{
+            const data: { user: number, question: number, report: string, report_details: string } = {
```

Comment:
make this an actual datatype too

Reply:
We fixed the data type!

Reply:
> Most of it seems okay, there just some efficiency and redundancy issues that need to be addressed.

We made the changes you requested, please let us know if we missed anything!